tools/pipelines-tasks/AVDAppAttachPublish/main.ts

```diff
@@ -0,0 +1,131 @@
+const TARGET_DLL = path.join(__dirname,'node_modules/common-helpers/lib/AppAttachFrameworkDLL/AppAttachKernel.dll');
+const TARGET_DLL_TELEMETRY = path.join(__dirname, 'node_modules/common-helpers/lib/AppAttachFrameworkDLL/AppAttachTelemetry.dll');
+
+function isNonEmpty(str: string): boolean {
+    return (!!str && !!str.trim());
+}
+
+function getResourceGroupNameFromUri(resourceUri: string): string {
+	if (isNonEmpty(resourceUri)) {
+		const parsedUrl = url.parse(resourceUri, true);
+
+		const pathname = parsedUrl.pathname || '';
+		const segments = pathname.split('/');
+
+		const resourceGroupIndex = segments.indexOf('resourceGroups');
+
+		if (resourceGroupIndex !== -1 && resourceGroupIndex < segments.length - 1) {
+			return segments[resourceGroupIndex + 1];
+		}
+	}
+	
+    return "";
+}
+
+async function run(): Promise<void> {
+	let isSuccessful = true;
+	try {
+		let connectedServiceName = tl.getInput('connectedServiceNameARM', true)!;
+
+		let subscriptionID: string = tl.getEndpointDataParameter(connectedServiceName, 'subscriptionid', true)!;
+
+		let storageAccountName = tl.getInput('storageAccount', true)!;
+		let azureEndpoint: AzureEndpoint = await new AzureRMEndpoint(connectedServiceName).getEndpoint();
+		let accessToken: string = (await azureEndpoint.applicationTokenCredentials.getToken()).toString();
+
+		const storageArmClient = new armStorage.StorageManagementClient(azureEndpoint.applicationTokenCredentials, (azureEndpoint.subscriptionID)!);
+		let storageAccount: StorageAccount = await storageArmClient.storageAccounts.get(storageAccountName);
+		let storageAccountResourceGroupName = getResourceGroupNameFromUri(storageAccount.id);
+
+		let accessKeys = await storageArmClient.storageAccounts.listKeys(storageAccountResourceGroupName, storageAccountName, null);
+		let accessKey: string = accessKeys[0];
+		let storageAccountConnectionString: string = `DefaultEndpointsProtocol=https;AccountName=${storageAccountName};AccountKey=${accessKey};EndpointSuffix=core.windows.net`;
+		let username: string = (process.env.BUILD_REQUESTEDFOREMAIL)!;
+		const PUBLISH_TO_AZURE_TASK = '8';
+
+		const appattachConfig = {
+			'emailId': username,
+			'accessToken': accessToken,
+			'azureStorageKey': storageAccountConnectionString,
+			'azureWorkspace': tl.getInput('workSpace', true),
+			'azureStorageFileShare': tl.getInput('fileShare', true),
+			'azureStorageAccount': storageAccountName,
+			'azureSubscriptionId': subscriptionID,
+			'azureResourceGroup': tl.getInput('resourceGroupName', true),
+			'azureHostPoolName': tl.getInput('hostPool', true),
+			'azureApplicationGroupName': tl.getInput('applicationGroup', true),
+			'taskType': PUBLISH_TO_AZURE_TASK,
+			'appAttachImagePath': tl.getInput('vhdxPath', true),
+			'clientType': helpers.CLIENT_TYPE,
+			'clientVersion': helpers.CLIENT_VERSION
+		};
+
+		const jsonString = JSON.stringify(appattachConfig);
+		tl.debug(jsonString);
```

Comment:
This is going to log the `accessToken`, which seems pretty dangerous. Sounds like something that could allow write access to the Azure resources to people who only have read access to the pipeline logs. Please remove that from the logs, unless you are completely sure that this access token is actually safe to disclose.

Reply:
Agreed, changing it